src/main/perl/Lock.pm

```diff
@@ -85,7 +86,10 @@ If a lock is set for the lock file, returns SUCCESS, undef otherwise.
 
 sub is_locked() {
   my $self=shift;
-  return SUCCESS if (-e $self->{'LOCK_FILE'});
+  if (-e $self->{'LOCK_FILE'}) {
+    my $fh=FileHandle->new("> ". $self->{'LOCK_FILE'});
+    return SUCCESS if (not flock($fh, LOCK_EX|LOCK_NB));
```

Comment:
You are locking the file momentarily in order to test that it is locked, then throwing the lock away again.  This is introducing a new race condition: a process could easily pass the first time `is_locked()` is called but fail the second time, causing the program to exit with an error.

Although you could use `fcntl` with `F_GETLK` to test for a lock without actually obtaining one, this is of dubious use as the lock could be gone before the `is_locked()` method returns.

It would be better to throw away the `is_locked()` method completely, and just have `set_lock()` attempt to run `flock()` multiple times with `LOCK_EX|LOCK_NB` until it wants to give up waiting any longer.